subsys/net/ip/canbus_socket.c

```diff
@@ -20,8 +20,9 @@ LOG_MODULE_REGISTER(net_sockets_can, CONFIG_NET_SOCKETS_LOG_LEVEL);
 
 enum net_verdict net_canbus_socket_input(struct net_pkt *pkt)
 {
-	if (net_pkt_family(pkt) == AF_CAN &&
-	    net_if_l2(net_pkt_iface(pkt)) == &NET_L2_GET_NAME(CANBUS_RAW)) {
+	__ASSERT_NO_MSG(net_pkt_family(pkt) == AF_CAN);
```

Comment:
Hmm, why this change? Asserts can be turned off so we do not then have the family check.

Reply:
As this function is not exposed in userspace it will only be called by the kernel itself. But then my change introduces family checks BEFORE calling family-specific socket APIs as it never makes sense to call `net_*some-family*_input()` methods for packets that do not belong to the corresponding family.

In other words: This check has not been removed but moved further upstream to the single (single) call site. I also do not see any reason why this method should ever be called elsewhere (and if so the assertion should surface the problem - which would be a kernel bug).

Hope this makes sense?